### ISSUE TITLE
Wrong public from private key extraction logic

### ISSUE THREAD

**Issue by dulanov** [nem-toolchain/nem-toolchain#126]
### Description

Address generated from given private key is not correct!

### Steps to reproduce

Generate a new account and try to import generated private key in NanoWallet:

```
> nem --chain testnet account vanity --no-estimate TABC TACB TBAC TBCA TCAB TCBA
Address: TABCSO-OLWKHI-VJNZAN-TAFLN7-TM4RNG-MPLBKJ-JAM3
Public key: ab0a9835f6b01cb6cbce6ea1b1c3c54a4bf6e5b9684d40c1f3dc6e329b82f3a6
Private key: c06faacfd991564a5f2e1910dffbdeb464b9fa06d83eec5a96c9708e93d7aab9
```

### Expected behavior

Receive the similar address in NanoWallet - `TABCSO-OLWKHI-VJNZAN-TAFLN7-TM4RNG-MPLBKJ-JAM3`

### Actual behavior

![image](https://user-images.githubusercontent.com/39793/31384195-24707a82-adbe-11e7-9856-491ea5ff73e1.png)

### Library version

`v0.2`

### Go version (e.g. `go version`)

`go version go1.9 windows/amd64`

### OS version (e.g. `uname -a`)

`MINGW64_NT-6.1 dulanov-PC 2.9.0(0.318/5/3) 2017-09-13 23:16 x86_64 Msys`

**Comment by dulanov** [comment]
After a quick research I found that a problem with `ed25519` go library, because the returned public key is wrong:

```
prv, _ := hex.DecodeString("2c52aee96f0e30f21c86b3fab7a18e927f579618818e8148e7ded1e01875ef0b")
pub, _, _ := ed25519.GenerateKey(bytes.NewReader(prv))
exp, _ := hex.DecodeString("9d1e9d01ab916dbdde0e76ba43df2246575d637db0bca090f46c1abce19a43e3")
assert.Equal(t, pub, KeyPair{Public: exp})
```

**Comment by r8d8** [comment]
Fixed in #86ada5ae92697ede13fc2f622a63cbfe366b6f5d
